coreclient/src/clients/process/process_qs.rs

```diff
@@ -804,3 +837,156 @@ async fn handle_message_edit(
 
     Ok(message)
 }
+
+/// A processor for the streamed QS events.
+///
+/// This processor is meant to be used in the streaming context where the events are streamed one
+/// by one and this process never finishes until the stream is closed. Each event is processed by
+/// `[Self::process_event]`.
+#[derive(Debug)]
+pub struct QsStreamProcessor {
+    core_user: CoreUser,
+    responder: Option<QsListenResponder>,
+    /// Accumulated but not yet processed messages
+    ///
+    /// Note: It is safe to keep messages in memory here, because they are not yet acked. In case,
```

Comment:
It's also safe because they have not yet been decrypted. Might be worth noting here.